scripts/cl_acquisition.py

```diff
@@ -60,120 +66,268 @@ def __init__(self):
+        # incorrect metadata.
+        pxs = self.get_pixel_size()
+
+        rep = self.repetition
+        center = (tl[0] + (pxs[0] * (rep[0] - 1)) / 2,
+                  tl[1] - (pxs[1] * (rep[1] - 1)) / 2)
+        logging.debug("Computed data width to be %s x %s, with center at %s",
+                      pxs[0] * rep[0], pxs[1] * rep[1], center)
+
+        return center, pxs
+
+    def assemble_sem_data(self, data: List[DataArray]) -> DataArray:
+        """
+        Assemble SEM data and metadata
+        :param data: sorted with X fast, Y slow. Each DataArray should be of shape (1,1).
+        :return: DataArray with the correct metadata and 5D shape (111YX)
+        """
+        # get metadata, no need to ask directly to the component because the metadata is already embedded in the first dataset
+        logging.debug("Assembling SEM data")
+        md = data[0].metadata.copy()
+        center, pxs = self.get_center_pxs(data[0])
+        md[model.MD_PIXEL_SIZE] = pxs
+        md[model.MD_POS] = center
+        md[model.MD_DESCRIPTION] = "Secondary electrons"
+
+        # Make a big array, and specify that the order is YX
+        full_data = model.DataArray(data, metadata=md)
+        xres, yres = self.repetition
+        # full_data = numpy.reshape(full_data, [1, 1, 1, yres, xres]) # fails if data wasn't of shape 1,1
+        full_data.shape = (1, 1, 1, yres, xres)  # fails if data wasn't of shape 1,1
+        return full_data
+
+    def assemble_opt_data(self, data: List[DataArray], sem_da: DataArray) -> DataArray:
+        """
+        Assemble optical data and metadata
+        :param data: sorted with X fast, Y slow. Each DataArray should contain a 1D image of shape (1, C)
+        :param sem_da: An already assembled DataArray of the same X&Y, used for the metadata information
+        :return: DataArray with the correct metadata and 5D shape (C11YX)
+        """
+        logging.debug("Assembling optical data")
+        md = data[0].metadata.copy()
+        md[model.MD_PIXEL_SIZE] = sem_da.metadata[model.MD_PIXEL_SIZE]
+        md[model.MD_POS] = sem_da.metadata[model.MD_POS]
+        md[model.MD_DESCRIPTION] = "CL spectrum"
+        md[model.MD_DIMS] = "CTZYX"
+
+        # Make a big array, and specify that the order is YX
+        full_data = model.DataArray(data, metadata=md)  # shape = Y*X, 1, C
+        full_data = full_data.swapaxes(0, 2)  # shape = C, 1, Y*X
+        xres, yres = self.repetition
+        full_data.shape = (full_data.shape[0], 1, 1, yres, xres)  # shape = C, 1, 1, Y, X
+
+        return full_data
+
+    def start_sem_acquisition(self):
+        """
+        Start SEM acquisition. It will keep acquiring (spot) data in the background
+        until complete_sem_acquisition() is called
+        """
+        self._sem_data_received.clear()
+        self._last_sem_data.clear()
+
+        self.sed.data.subscribe(self.on_sem_data)
+
+    def on_sem_data(self, df, da):
```

Comment:
```suggestion
    def on_sem_data(self, _, da):
```
Maybe also add small docstring?

Reply:
I prefer to leave the name of the argument, as it has to follow the specific interface. It's better for the docstring. (which I've added).